ui/src/components/credentials/CredentialDetails.tsx

```diff
@@ -276,7 +294,7 @@ export function CredentialDetails() {
       )}
       {isAsyncTaskDataAvailable(credential) && showRevokeModal && (
         <CredentialRevokeModal
-          credential={credential.data}
+          credential={{ ...credential.data, revoked }}
```

Comment:
in case the status is not published, getRevocationStatus returns false but getCredentials list returns item with revoked: true value, so to display correct view on detail page i set query parameter with correct revoked status and also add tooltip to revoke button which shows not published status.